scripts/link-check-git-diff.sh

```diff
@@ -1,13 +1,21 @@
 #!/usr/bin/env bash
 set -euo pipefail
 
-exclude="${CHECK_LINKS_EXCLUDE_LIST:-$(dirname $0)/exclude-links.txt}"
+repo="$(dirname "$(realpath "$(dirname "$0")")")"
+pushd "$repo"
+
+exclude="${CHECK_LINKS_EXCLUDE_LIST:-$(dirname "$0")/exclude-links.txt}"
 differ="git diff $(git merge-base HEAD origin/master)"
-changed="$differ --name-only :^redirects-list.json"
+changed="$($differ --name-only -- . ':!'"$exclude" ':!redirects-list.json')"
```

Comment:
```suggestion
changed="$($differ --name-only -- '*.md' '*.js')"
```

@shcheklein this would address the missing checkbox in this PR. Seems a better idea.

Reply:
may be just *.md? why are we testing sources?

Reply:
it's definitely better to keep it consistent!

Reply:
the link-check has always tested `js` sources (some of the entries in the exclusions list are for `js` sources). Seems logical if there are URLs appearing in `js` code.

Reply:
makes sense, @casperdcl, just a bit worried about the exclusion file growing infinitely and containing old record as we go .. if we include code we now should be including `*.ts` and `*.tsx`, `js` is only a small portion of the code base now.

Reply:
Ah I see those extensions have been added recently. Should add them here too, really.

Was also concerned about dead exclusions but think that can be solved separately using a script.

Reply:
also, not sure if we want to include css btw? they might contain some links as well. WDYT?

Reply:
Regarding that script... just automatically found these:

```
http://localhost:3000/
http://millionsongdataset.com/pages/getting-dataset/
http://ogp.me/ns
https://blog.$
https://dvc.org/img/<filename>.gif
https://dvc.org/uploads/images/2020-02-10/image.png
https://example.com/path/to/data
https://remote.dvc.org/dataset-registry/a3/04af...
https://s3-us-east-2.amazonaws.com/dvc-s3-repo/
```

Looks like we can remove them? Can also add the script to check for unused exclusions to the CI and pre-commit hook.

Reply:
Opened #1184

Reply:
> also, not sure if we want to include css btw? they might contain some links as well. WDYT?

`finder $(git ls-files '*.css')` comes up blank :)

Reply:
👍

Reply:
so, let's add them anyway - it's good that we don't have anything yet :) let's keep it clean from the very beginning

Reply:
ok should be fine to merge now.